src/blockdata/transaction.rs

```diff
@@ -616,6 +616,37 @@ pub enum SigHashType {
     /// 0x83: Sign one output and only this input (see `Single` for what "one output" means)
     SinglePlusAnyoneCanPay	= 0x83
 }
+serde_string_impl!(SigHashType, "a SigHashType data");
+
+impl fmt::Display for SigHashType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = match self {
+            SigHashType::All => "SIGHASH_ALL",
+            SigHashType::None => "SIGHASH_NONE",
+            SigHashType::Single => "SIGHASH_SIGNLE",
+            SigHashType::AllPlusAnyoneCanPay => "SIGHASH_ALL | SIGHASH_ANYONECANPAY",
```

Comment:
IMO we should drop the spaces here which might make parsing easier in some contexts (e.g. shell ... though of course shell will also barf on the `|` ;))